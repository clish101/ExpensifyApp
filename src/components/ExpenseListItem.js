import React from 'react';

import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, dispatch, id }) => (
  <div>
    <p>These are what was spend on</p>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    amount - createdAt
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
