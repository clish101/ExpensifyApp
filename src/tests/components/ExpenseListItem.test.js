import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';
import React from 'react';
import { shallow } from 'enzyme';

test('Should render expense list with expenses', () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
  expect(wrapper).toMatchSnapshot();
});
