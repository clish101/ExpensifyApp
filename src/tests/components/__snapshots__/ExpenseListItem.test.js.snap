// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense list with expenses 1`] = `
<div>
  <p>
    These are what was spend on
  </p>
  <Link
    to="/edit/2"
  >
    <h3>
      Rent
    </h3>
  </Link>
  amount - createdAt
  <p>
    109500
     - 
    -345600000
  </p>
</div>
`;
